tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/MicroSecEndTest.java

```diff
@@ -0,0 +1,468 @@
+
+            for (var vertex : transposeFlowGraph.getVertices()) {
+                hasGateway(violationsSet, variant, vertex);
+
+                hasAuthenticatedRerquest(violationsSet, variant, vertex);
+
+                hasOptionalAuthorizedEntrypoint(violationsSet, variant, vertex, flowGraph.getTransposeFlowGraphs());
+
+                hasTransformedEntryIdentity(violationsSet, variant, vertex);
+
+                hasTokenValidation(violationsSet, variant, vertex);
+
+                hasLoginAttemptsRegulation(violationsSet, variant, vertex);
+
+                hasEncryptedEntryConnection(violationsSet, variant, vertex);
+
+                hasEncrytedInternalConnection(violationsSet, variant, vertex);
+
+                hasLocalLogging(violationsSet, variant, vertex);
+
+                hasLogSanitization(violationsSet, variant, vertex);
+
+                hasOptionalMessageBroker(violationsSet, variant, vertex, flowGraph.getTransposeFlowGraphs());
+
+            }
+        }
+        checkCrossTransposeFlowGraphViolations(flowGraph, violatingTransposeFlowGraphs, violationsSet);
+
+    }
+
+    @Test
+    void testConstraints() {
+        for (var model : TUHH_MODELS.keySet()) {
+            for (int variant : TUHH_MODELS.get(model)) {
+                Set<Integer> violationSet = new TreeSet<Integer>();
+                String variationName = model + "_" + variant;
+                performAnalysis(Paths.get(location, model, variationName)
+                        .toString(), variant, violationSet);
+                logger.info("Variant: " + variationName);
+                logger.info("Violations: " + violationSet);
+                assertFalse(violationSet.contains(variant));
+            }
+        }
+    }
+
+    @Test
+    void testDetectCycles() {
+        var model = Paths.get(location, "anilallewar", "anilallewar_0")
+                .toString();
+
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+
+        assertTrue(flowGraph.wasCyclic());
+
+    }
+    
+    
+    @Test
+    void testRamCopyIssue() {
+        var model = Paths.get(location, "sqshq", "sqshq_18")
+                .toString();
+        Set<Integer> numbers = new HashSet<>();
+        
+        for(int i = 0; i < 50; i++) {
+            var analysis = buildAnalysis(model);
+            analysis.initializeAnalysis();
+            var flowGraph = analysis.findFlowGraphs();
+            flowGraph.evaluate();
+            var numVertex = 0;
+            for(var tfg: flowGraph.getTransposeFlowGraphs()) {
+                numVertex+=tfg.getVertices().size(); 
+            } 
+            numbers.add(numVertex);
+
+
+            logger.info("Num TFGS: " + flowGraph.getTransposeFlowGraphs().size()+ " resulting in numVertex:"+ numbers);
+
+            assertEquals(numbers.size(),1);
+        }
+    }
+    
+    @Test
+    void simpleLoopCheck() {
```

Comment:
Does this test belong in this class? It doesn't check anything TUHH related and deals more with the cyclic finder

Reply:
Agree we could move it to the online shop test or move it to its own class, what would you prefer?

Reply:
Could you give it its own class? Named something like `CyclicFinderTest`. Will come in handy if we add more pure cyclic finder tests